### ISSUE TITLE
TFLite conversion bug in TF1

### ISSUE THREAD

**Issue by Tombana** [larq/compute-engine#44]
These are some notes and thoughts on tflite conversion.

When converting models, we could put either `tf.sign` or `lqce.bsign` in the graph. It doesn't matter which you choose because both of these become custom ops ("Sign" or "Bsign") and in tflite we can handle them appropriately. We can even register both strings in tflite and refer them to the same op.

There is still a weird issue about data types: in BinaryAlexNet near the end there is
BatchNorm -> Flatten -> Sign -> Dense

When using the tflite converter from tensorflow 2, this is all fine and the Flatten becomes a Reshape op.

When converting using tensorflow1 conversion, the flatten layer becomes a very weird thing with ( "Shape" , "StridedSlice" , "Pack" ) and a shortcut. Even though this should not be there, it revealed an issue:
When putting `tf.sign` after this flatten layer, the "Shape" op would go from float32 to int32.
When putting `tf.bsign` after this flatten layer, the "Shape" op would go from float32 to float32, yielding an error because "StridedSlice" expected an int32.

So this shows that our `bsign` op is somehow different from `tf.sign` regarding type info.
Maybe we should try to investigate this difference.
As suggested by Arash, maybe we can try putting the bsign dtypes in a different order.
